### PR TITLE
Edit Selector Implementation to be concise

### PR DESCRIPTION
**Pull request by dollinad**
Edit Selector Implementation in order to be concise.

Changes involve:
- Sentence rewrites 
- Removal of dead links
- Correction of grammar and spelling

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _changes were made to docs file only__

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23049)

### CODE REVIEW

File: docs/components/style.md [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-The style system is generic over quite a few things, in order to be shareable
-with Servo's layout system, and with [Stylo][stylo], an ambitious project that
-aims to integrate Servo's style system into Gecko.
+In order to be sharable and compatible with [Stylo][stylo](a project that aims 
+to integrate Servo's style system into Gecko), the sytle must be consistent.
 
-The main generic trait is [selectors' SelectorImpl][selector-impl], that has all
-the logic related to parsing pseudo-elements and other pseudo-classes appart
-from [tree-structural ones][tree-structural-pseudo-classes].
+The consitency is implemented in [selectors' SelectorImpl][selector-impl],
```
Comment (jdm) [inline docs/components/style.md]: s/consitency/consistency/.

File: docs/components/style.md [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
 Boris Zbarsky and Patrick Walton about how style sharing works.
 
 <a name="selector-impl"></a>
 ## Selector Implementation
 
-The style system is generic over quite a few things, in order to be shareable
-with Servo's layout system, and with [Stylo][stylo], an ambitious project that
-aims to integrate Servo's style system into Gecko.
+In order to be sharable and compatible with [Stylo][stylo](a project that aims 
+to integrate Servo's style system into Gecko), the sytle must be consistent.
```
Comment (jdm) [inline docs/components/style.md]: s/sytle/style/

**Review by jdm — APPROVED** [review]
These are nice improvements. Thanks!

File: docs/components/style.md (hunk 1 above)
Comment (dollinad) [reply docs/components/style.md]: Updated!:)

File: docs/components/style.md (hunk 2 above)
Comment (dollinad) [reply docs/components/style.md]: Updated!

**Comment by jdm** [comment]
Could you squash these commits into one?

**Comment by dollinad** [comment]
Absolutely

**Comment by jdm** [comment]
That seems to have left 3 commits.

**Comment by dollinad** [comment]
There should be 1 now:)

**Pull request merged**